WhisperAPI/WhisperAPI/Controllers/ContextController.cs

```diff
@@ -21,18 +23,18 @@ public ContextController(IContexts contexts)
 
         public override void OnActionExecuting(ActionExecutingContext actionExecutingContext)
         {
-            var searchQuery = (SearchQuery)actionExecutingContext.ActionArguments["searchQuery"];
+            Query query = actionExecutingContext.ActionArguments.Values.OfType<Query>().FirstOrDefault();
```

Comment:
Why did you changed this line ? Simply changing the cast type to (Query) make all the tests pass... are missing some coverage in the tests ?
Because right now the tests fail because somethime we don't have any ActionArguments so we are calling ".Values" on a null object

Reply:
I found the problem. In the tests the actionExecutingContext is mocked to return the value on this explicit call : 
"actionExecutingContext.ActionArguments["searchQuery"]" the value isn't actually contained inside. So your solution might work here but you'll need to check for null i.e. : ".Values?.OfType<Query>()" AND to change the mock in all the failling tests. OR you could do something that can try to extract "searchQuery" then "SelectQuery" if the first one fail (might be ugly). OR we could rename "searchQuery" and "selectQuery" for "query" (Do we explictily need then to be named that way ?). 
Don't hesitate to ping me if you have questions about this comment